chainer/datasets/image_dataset.py

```diff
@@ -118,7 +118,7 @@ def get_example(self, i):
         path, int_label = self._pairs[i]
         full_path = os.path.join(self._root, path)
         with Image.open(full_path) as f:
-            image = numpy.asarray(f, dtype=self._dtype)
+            image = numpy.asarray(f.convert('RGB'), dtype=self._dtype)
```

Comment:
For an input grey-scale image (H, W), this solution will output a (3, H, W) ndarray. However as stated here https://github.com/pfnet/chainer/blob/master/chainer/datasets/image_dataset.py#L24 we'd like to output a (1, H, W) for a grey-scale input. How about `image = image[:, :, numpy.newaxis]` if the input is 2D?